src/Bicep.Core.IntegrationTests/ExtensibilityTests.cs

```diff
@@ -656,5 +656,240 @@ extension foo as foo
+            var moduleAUri = new Uri("file:///modulea.bicep");
+
+            // TODO(kylealbert): Remove 'with' clause in template when that's removed
+            // TODO(kylealbert): Uncomment graph when I figure out how to deal with the registry.
+            var files = new Dictionary<Uri, string>
+            {
+                [paramsUri] =
+                    """
+                    using 'main.bicep'
+
+                    param inputa = 'abc'
+
+                    extension k8s with {
+                      kubeConfig: 'abc'
+                      namespace: 'other'
+                    }
+                    """,
+                [mainUri] =
+                    """
+                    param inputa string
+
+                    extension kubernetes with {
+                      kubeConfig: 'DELETE'
+                      namespace: 'DELETE'
+                    } as k8s
+
+                    //extension 'br:mcr.microsoft.com/bicep/extensions/microsoftgraph/v1.0:0.1.8-preview'
+
+                    module modulea 'modulea.bicep' = {
+                      name: 'modulea'
+                      params: {
+                        inputa: inputa
+                      }
+                      extensionConfigs: {
+                        kubernetes: {
+                          kubeConfig: 'fromModule'
+                          namespace: 'other'
+                        }
+                      }
+                    }
+
+                    output outputa string = modulea.outputs.outputa
+                    """,
+                [moduleAUri] =
+                    """
+                    param inputa string
+
+                    extension kubernetes with {
+                      kubeConfig: 'DELETE'
+                      namespace: 'DELETE'
+                    }
+
+                    //extension 'br:mcr.microsoft.com/bicep/extensions/microsoftgraph/v1.0:0.1.8-preview' as graph
+
+                    output outputa string = inputa
+                    """
+            };
+
+            var compilation = ServicesWithModuleExtensionConfigs.BuildCompilation(files, paramsUri);
+
+            compilation.Should().NotHaveAnyDiagnostics_WithAssertionScoping(d => d.IsError());
+        }
+
+        [DataTestMethod]
+        [DataRow("MissingExtensionConfigsDeclaration")]
+        [DataRow("MissingRequiredExtensionConfig")]
+        [DataRow("MissingRequiredConfigProperty")]
+        [DataRow("PropertyIsNotDefinedInSchema")]
+        [DataRow("ConfigProvidedForExtensionThatDoesNotAcceptConfig")]
+        public void Module_with_invalid_extension_config_produces_diagnostic(string scenario)
+        {
+            var mainUri = new Uri("file:///main.bicep");
+            var moduleAUri = new Uri("file:///modulea.bicep");
+
+            var extensionConfigsStr = scenario switch
+            {
+                "MissingExtensionConfigsDeclaration" => "",
+                "MissingRequiredExtensionConfig" => "extensionConfigs: {}",
```

Comment:
We can move the config strings, diagnostic error codes and messages to `DataRow` as well.

Reply:
Done.